lib/api/api.go

```diff
@@ -1652,6 +1681,16 @@ func toJsonFileInfoSlice(fs []db.FileInfoTruncated) []jsonFileInfoTrunc {
 	return res
 }
 
+func toJsonPendingFolderMap(folders map[string]map[protocol.DeviceID]db.ObservedFolder) map[string]interface{} {
+	res := make(map[string]interface{}, len(folders))
+	for id, devices := range folders {
+		res[id] = map[string]interface{}{
+			"offeredBy": devices,
```

Comment:
No strong reason.  Just easier to grasp if the relation of device IDs to folder IDs is given a name.

Do you think that's not worth the extra code?

Reply:
In addition, `offeredBy` used to be an array, so translation was actually necessary.  Was changed to an object for a forgotten reason, but I think that's preferable.

Reply:
Yes I'd drop it, I think this is pretty self-evident respectively I can't come up with any way to misinterpret it (which doesn't say much). I don't feel strongly about this though and am fine with either.

Reply:
If we ever want to add other information on the folder level, then it makes sense again.  That's what I'm planning for one of the "candidate" devices endpoints, so there's some consistency argument coming up at least.

Reply:
Then instead of a double map I'd use `map[string]PendingFolder`, where `observedBy` is a member. Then you can have json tags on that and thus don't need a custom json converter function.

Reply:
Done.  Something like that I already had on the candidate devices branch, so again happy to reach some more consistency.

I don't think we need to apply the same to `PendingDevices` as there is definitely no nesting needed?